### ISSUE TITLE
sec.crypt.passphrase_file nothing mentioned in documentation about how to set it.

### ISSUE THREAD

**Issue by tya99** [weechat/weechat#1286]
## Bug summary
No apparent way to apply data with passphrase file. ie [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file)

So the idea was I wanted to set a secure [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file) on external medium like a encrypted usb stick that is put into my computer when I sit down.

In this case I would use a `passphrase_file` but not a `passphrase`

I set the [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file) but the data in `sec.conf` was still unencrypted.

Looking at the [/secure](https://weechat.org/files/doc/stable/weechat_user.en.html#command_weechat_secure) command it appears that there is nothing mentioned about it other than you may set `WEECHAT_PASSPHRASE` - which I have not done.

## Suggested solutions
1. Update documentation so that it is apparent how to apply the encryption when using [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file) **without a passphrase**. I am assuming the way this currently works is it's required in addition to a passphrase.
2. Change the code, if you can store the data in `sec.conf` unencrypted and with a `passphrase` and `passphrase_file` surely you can store it with just a `passphrase_file`.
3. Remove the [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file) feature if it isn't even there, obviously the least desirable solution as it sounds like a great feature to have.

## Expected Results
1. The most reasonable thing I think would be a command line option like `/secure passphrase_file on` and `/secure passphrase_file off` which would encrypt the data with the `passphrase_file`.

2. Alternatively if someone wanted to use a `passphrase` and a `passphrase_file` something like `/secure  passphrase <passphrase>|-passphrase_file` would make sense. `decrypt <passphrase>|-discard` could still be used to remove the `passphrase` and `passphrase_file`.

3. If either of these commands are run without [sec.crypt.passphrase_file](https://weechat.org/files/doc/stable/weechat_user.en.html#option_sec.crypt.passphrase_file) being set the user should be told that no `passphrase_file` is set.

**Comment by flashcode** [comment]
The process to use a passphrase protecting and read it from a file on startup your data is:

1. define a passphrase: `/secure passphrase xxxxxx`
2. add some data: `/secure add name xxxx`
3. store the passphrase in a file, and `/set sec.crypt.passphrase_file /path/to/passphrase.txt` (the file must contain only the passphrase, a newline at the end is allowed and ignored)

I just tested that, it works fine for me.

So could you please give some info about your problem? When you define the file with passphrase, what's the error on next WeeChat startup?

**Comment by weechatter** [comment]
Works fine for me too, but i suggest to add more information to /help secure

**Comment by tya99** [comment]
I think I misunderstood this feature however now that I understand I can still use it. The initial trouble was I had created a `passphrase_file` with `dd` ie:

```
dd if=/dev/urandom of=passphrase_file bs=512 count=8
8+0 records in
8+0 records out
4096 bytes (4.1 kB, 4.0 KiB) copied, 0.000319749 s, 12.8 MB/s
```

Of course the contents of that would be binary and couldn't be specified on the terminal. That's why I thought there might be some way to read in a file for encryption.

`cat passphrase_file | base64 | tr -d '\n' > passphrase_file_base64` solves that issue though.

I then copied the 6586 characters to my clipboard and used:
`/secure passphrase -delete`
`/secure set passphrase <base_64_secret>`
`/set sec.crypt.passphrase_file ~/passphrase_file_base64`
